### PR TITLE
PE-153 skip release branches

### PR DESCRIPTION
**Pull request by Romack**

### CODE REVIEW

File: __tests__/utils.test.ts [hunk 1]
```diff
@@ -34,6 +34,17 @@ describe('shouldSkipBranchLint()', () => {
     expect(shouldSkipBranchLint('chore/add-all-contributors-OSS-102')).toBe(false);
   });
 
+  it('should skip commonly used branch patterns', () => {
+    expect(shouldSkipBranchLint('main')).toBe(true);
+    expect(shouldSkipBranchLint('master')).toBe(true);
+    expect(shouldSkipBranchLint('production')).toBe(true);
+    expect(shouldSkipBranchLint('gh-pages')).toBe(true);
+    expect(shouldSkipBranchLint('release/v1')).toBe(true);
```
Comment (Romack) [inline __tests__/utils.test.ts:42]: question: should we remove support for the branches that start with `release/v`?

It appears that in this meeting, we decided that release branches should be in the format of`release/<major>.<minor>`
https://invitationhomes.atlassian.net/wiki/spaces/TECH/pages/1770651649/2023-06-23+Continuous+Delivery+SIG+Meeting

Comment (andyasberry) [reply __tests__/utils.test.ts:42]: great question and good catch - my vote is to only support what we've decided to support 🗳️ 

on the other hand, should we allow PRs to release branches to skip jira-lint? i think that's how jira-lint came when we forked it, but seems like we'd still want a jira story even if we're trying to merge to a release branch.

File: src/constants.ts [hunk 2]
```diff
@@ -19,6 +19,7 @@ export const DEFAULT_BRANCH_PATTERNS: RegExp[] = [
   /^production$/,
   /^gh-pages$/,
   /^release\/v(\d+\.)?(\d+\.)?(\d+)$/,
+  /^release\/(\d+\.)(\d+)$/,
```
Comment (andyasberry) [inline src/constants.ts:22]: **suggestion: (non-blocking)** if you like, we can remove the parentheses, i think (i'm good with whatever you prefer):

```suggestion
  /^release\/\d+\.\d+$/,
```

**Review by andyasberry — COMMENTED** [review]
Looks good to me!

Since this is an action, and since it doesn't have the slick ESBuild pre-commit hooks we sest up in the TS GH Action Template, we have to build the output and commit that (this is fairly normal for actions out in the wild).

**Review by andyasberry — APPROVED** [review]

**Pull request closed without merging**